### DISCUSSION TITLE
.local-chromium: No such file or directory

### DISCUSSION THREAD

**Discussion by Jam0r85** [spatie/browsershot#746] [Q&A]
I've provisioned a new server on Hetzner (Ubuntu 22.04) through Forge and when trying to install using the guide I get the following error -

`chmod: cannot access '/usr/lib/node_modules/puppeteer/.local-chromium': No such file or directory`

I've looked in this folder and the only files/folders present are

![Screenshot 2023-06-29 at 12 50 00](https://github.com/spatie/browsershot/assets/18726355/80faa600-77e9-42a7-9bbb-e2a884f0201d)

I can't find any reference to this error elsewhere. Even tried creating a new server with different hardware and I have the same issue.

Installed chromium directly and it's located at '/snap/bin/chromium' but even when setting the path to this I still get corrupted files.

Everything works fine locally so it must be a server issue.

Any ideas?

**Comment by aleksandertabor** [comment]
Install Puppeteer v17:
`sudo npm install -g puppeteer@17 --unsafe-perm=true`

Higher versions don't work.

**Reply by Yiddishe-Kop** [reply]
This fixed it for me.

But v17 isn't supported anymore. From looking in to this - the issue seems to be a misconfigured cache path, which since puppeteer v19 was changed to `~/.cache/puppeteer` ([source](https://pptr.dev/guides/configuration#changing-the-default-cache-directory)).

Maybe this package should be updated to support this 🤷‍♂️.

**Reply by sammyaxe** [reply]
did you manage to solve this?

**Reply by hyphen81** [reply]
also wondering if you were able to get this working without reverting to v17?

**Reply by richpeers** [reply]
bump

**Comment by AndresReyesDev** [comment]
To anyone having this problem, fix it in forge by upgrading puppeteer from version 17 to the latest version (21 at the time), then use this command to get the chrome version specific to this version of puppeteer

`npx @puppeteer/browsers install chrome@121.0.6167.85`

When it finishes installing chrome it will give the path that you can use in your Laravel project using the parameter:

```
Browsershot::html('Foo')
  ->setChromePath("/path/to/my/chrome")
```

Note, you need NodeJS 20 to use the latest version of Puppeteer.

Important info:
[Puppeteer Chromium Support](https://pptr.dev/chromium-support)
[Google: How can I get Chrome for Testing binaries?](https://developer.chrome.com/blog/chrome-for-testing/#how_can_i_get_chrome_for_testing_binaries)
[Browsershot Requirements](https://spatie.be/docs/browsershot/v3/requirements#content-custom-chromechromium-executable-path)
If you use Forge and want to upgrade NodeJS:
[Forge Cookbook](https://forge.laravel.com/docs/servers/cookbook.html#upgrading-node-js)
